pilot/pkg/model/service.go

```diff
@@ -45,23 +45,47 @@ type Service struct {
 	// Hostname of the service, e.g. "catalog.mystore.com"
 	Hostname string `json:"hostname"`
 
-	// Address specifies the service IPv4 address of the load balancer
+	// Address specifies the virtual IP address associated with the service.
+	// If empty
 	Address string `json:"address,omitempty"`
 
 	// Ports is the set of network ports where the service is listening for
 	// connections
 	Ports PortList `json:"ports,omitempty"`
 
-	// ExternalName is only set for external services and holds the external
-	// service DNS name.  External services are name-based solution to represent
-	// external service instances as a service inside the cluster.
-	ExternalName string `json:"external"`
-
 	// ServiceAccounts specifies the service accounts that run the service.
 	ServiceAccounts []string `json:"serviceaccounts,omitempty"`
 
-	// LoadBalancingDisabled indicates that no load balancing should be done for this service.
-	LoadBalancingDisabled bool `json:"-"`
+	// MeshExternal (if true) indicates that the service is external to the mesh.
+	// These services are defined using Istio's ExternalService spec.
+	MeshExternal bool
+
+	// Virtual (if true) indicates that the service is not discoverable or directly
+	// routable using its hostname. These are typically used to indicate subsets of
+	// a given service (e.g., v1.reviews.bookinfo.com, where v1 is the name of a subset that
+	// indicates the set of instances associated with version v1 of the reviews service.
+	VirtualService bool
+
+	// Labels is a set of key-value pair metadata associated with each service instance
+	// associated with this service. For virtual services, the set of labels will be a super set
+	// of the set of labels associated with the respective parent service.
+	Labels Labels
+
+	// SelectionMode indicates how the proxy will identify the destination service for a given
+	// connection or request using the information available. HTTP based services will typically
+	// use VirtualHost based selection. TCP based services can be delineated using IP/CIDR blocks
+	// and some services can be identified solely using the port on which the connection arrives
+	// (typically used to proxy HTTPS traffic to external services, or by TCP services on platforms
+	// that do not have virtual IPs for services).
+	SelectionMode SelectionMode
+
+	// Resolution indicates how the service instances need to be resolved before routing
+	// traffic. Most services in the service registry will use static load balancing wherein
+	// the proxy will decide the service instance that will receive the traffic. External services
+	// could either use DNS load balancing (i.e. proxy will query DNS server for the IP of the service)
+	// or use the passthrough model (i.e. proxy will forward the traffic to the network endpoint requested
+	// by the caller)
+	Resolution Resolution
```

Comment:
Keep in mind we want to use the external service for mesh expansion too ( so we get label support )... I'm not sure what clientsideLB does - if it calls EDS it is probably ok

Reply:
clientsideLB is eds..